File: CPP/ByTier/D/2096_D_Wonderful_Lightbulbs.cpp
// Problem: CF 2096 D - Wonderful Lightbulbs
// https://codeforces.com/contest/2096/problem/D

/*
D. Wonderful Lightbulbs
Algorithm: XOR-based solution using properties of the operation.
Techniques: Bit manipulation, XOR, Gaussian elimination concept (implicit).
Time Complexity: O(n) per test case.
Space Complexity: O(1) excluding input.

The key insight is that each operation toggles 4 bulbs in a specific pattern.
By analyzing how these operations interact with XOR, we can deduce the original
position of the treasure using bitwise operations on coordinates.

The algorithm uses:
- XOR of all x-coordinates to find a component of the answer.
- XOR of (x + y) for all coordinates to get another relation.
- From there, compute one possible (s, t) such that s ^ x = s and t ^ (x + y) = t + s.

The operations are defined as toggling:
(x, y), (x, y+1), (x+1, y-1), (x+1, y)

Because each operation affects 4 positions, we can solve the problem
using XOR properties to determine a consistent and valid position of treasure.
*/

#include <stdio.h>

void solve() {
  int n;
  scanf("%i", &n);
  int a = 0, b = 0;
  for (int x, y, i = 0; i < n; ++i) {
    scanf("%i%i", &x, &y);
    a ^= x;              // XOR of all x coordinates
    b ^= (x + y);        // XOR of (x + y) for each coordinate
  }
  printf("%i %i\n", a, b - a);  // Output the possible treasure position
}

int main() {
  int t;
  scanf("%i", &t);
  while (t--)
    solve();
  return 0;
}


// https://github.com/VaHiX/codeForces/

File: Python/ByRound/2096/2096_D_Wonderful_Lightbulbs.py
# Problem: CF 2096 D - Wonderful Lightbulbs
# https://codeforces.com/contest/2096/problem/D

"""
D. Wonderful Lightbulbs

Problem Description:
We are given a configuration of lightbulbs that are ON in an infinite grid.
Each operation switches the state of four adjacent bulbs in a specific pattern:
(x, y), (x, y+1), (x+1, y-1), (x+1, y)
This operation can be thought of as a XOR operation on the bulb states.
We need to find one possible initial position of the treasure (a bulb that was originally turned ON).

Key Insight:
Since operations are reversible and each operation toggles 4 bulbs, we can think of this in terms
of XOR properties. The final configuration can be represented as a combination of these operations.

Approach:
Use XOR operations on x and x+y coordinates to compute the possible initial position.
The key idea is that for each bulb at (x_i, y_i), we track the XOR of x and the XOR of (x + y),
which leads us to determine a valid (s, t) where s ^= x and c ^= x + y.
Eventually, the result gives one valid solution (s, c - s).

Time Complexity: O(n)
Space Complexity: O(1)

This problem is about finding a consistent initial state by exploiting properties of XOR operations
on patterns of toggling bulbs in a grid.
"""

import sys

input = sys.stdin.readline
t = int(input())
for _ in range(t):
    n = int(input())
    s = 0   # XOR of all x coordinates
    c = 0   # XOR of (x + y) for all coordinates
    for _ in range(n):
        x, y = map(int, input().split())
        s ^= x           # Accumulate XOR of x values
        c ^= x + y       # Accumulate XOR of (x + y) values
    print(s, c - s)      # Output the computed candidate position


# https://github.com/VaHiX/codeForces/